fs/src/io_uring/memory.rs

```diff
@@ -72,11 +44,15 @@ pub struct PageAlignedMemory {
 }
 
 impl PageAlignedMemory {
-    fn alloc_huge_table(memory_size: usize) -> Result<Self, AllocError> {
+    fn alloc(memory_size: usize, use_huge_page: bool) -> Result<Self, AllocError> {
         let page_size = Self::page_size();
         debug_assert!(memory_size.is_power_of_two());
         debug_assert!(page_size.is_power_of_two());
-        let aligned_size = memory_size.next_multiple_of(page_size);
+        let aligned_size = if use_huge_page {
```

Comment:
This now goes against how this struct was designed or at least its name - we might no longer allocate a _page aligned_ memory. I guess for this PR we don't really need to introduce this condition, when somehow the provided input is not aligned, we will just allocate a bit more.